cranelift/codegen/src/machinst/lower.rs

```diff
@@ -775,6 +777,19 @@ impl<'func, I: VCodeInst> Lower<'func, I> {
             let loc = self.srcloc(inst);
             self.finish_ir_inst(loc);
 
+            // maybe insert random instruction
+            if ctrl_plane.get_decision() {
+                if ctrl_plane.get_decision() {
+                    let imm: u64 = ctrl_plane.get_arbitrary();
+                    let regs = self.alloc_tmp(crate::ir::types::I64);
+                    I::gen_imm_u64(imm, regs.regs()[0]).map(|inst| self.emit(inst));
+                } else {
+                    let imm: f64 = ctrl_plane.get_arbitrary();
+                    let regs = self.alloc_tmp(crate::ir::types::F64);
```

Comment:
The floats aren't working yet, I still have to figure out how to properly use a floating point register.